sdk/core/AzureCore/Source/Pipeline/Transport/HTTPHeader.swift

```diff
@@ -26,142 +26,350 @@
 
 import Foundation
 
-/// A dictionary of `HTTPHeader` values.
+// swiftlint:disable function_body_length type_body_length cyclomatic_complexity
+
+/// Type alias for HTTP header dictionary
 public typealias HTTPHeaders = [String: String]
+/// Extensions to work with `HTTPHeader` values within a collection of `HTTPHeaders`.
 
 /// Common HTTP headers.
-public enum HTTPHeader: String {
+public enum HTTPHeader: RequestStringConvertible, Equatable, Hashable {
+    /// Use when the header value you want is not in the list.
+    case custom(String)
     /// Accept
-    case accept = "Accept"
+    case accept
```

Comment:
Can we use associated value defaults here, e.g. `case accept(requestString: String = "Accept")`?

Then we could remove any special knowledge inside the getter for `requestString`, and also in the `init` we can just loop through all the cases and find one whose requestString matches (or return .custom otherwise) rather than having to do a big switch on string values there.

Reply:
If you do that, you would be able to override the value (`.accept("MadeUpValue"`), which at least in my mind would break the principle of least surprise.  The switch statements are annoying, but they are also fairly trivial to generate.

Reply:
That's true, it is ugly but I'm fine with dealing with that if it's auto-generated. Maybe just add some comments to indicate that the file should not be hand-modified, or to be careful if doing so.

Reply:
This file isn't auto-generated since it's in Core, but ones generated by the Swagger (where modelAsString is true) would be. Here, you would need to add a few annotations when adding new convenience values to AzureCore, but that's probably not going to happen with regularity. The compiler would force you to update the requestString method and, if you forgot to add it to the string init it would just consider it a custom value and everything would continue to work.